DocumentDataAPI/DocumentDataAPI/Data/Repositories/BiasSchema/NpgBiasPoliticalPartiesRepository.cs

```diff
@@ -0,0 +1,109 @@
+
+public class NpgBiasPoliticalPartiesRepository : IBiasPoliticalPartiesRepository
+{
+    private readonly IDbConnectionFactory _connectionFactory;
+    private readonly ILogger<NpgBiasPoliticalPartiesRepository> _logger;
+    private readonly ISqlHelper _sqlHelper;
+
+    public NpgBiasPoliticalPartiesRepository(IDbConnectionFactory connectionFactory, ILogger<NpgBiasPoliticalPartiesRepository> logger, ISqlHelper sqlHelper)
+    {
+        _connectionFactory = connectionFactory.WithSchema(Options.DatabaseOptions.Schema.Bias);
+        _logger = logger;
+        _sqlHelper = sqlHelper;
+    }
+
+    public async Task<BiasPoliticalPartiesModel?> Get(int partyId)
+    {
+        _logger.LogDebug("Retrieving political party with id {id} from database", partyId);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.QueryFirstOrDefaultAsync<BiasPoliticalPartiesModel>($"select * from political_parties where {BiasPoliticalPartiesMap.Id} = @Id", new { id = partyId });
+    }
+
+    public async Task<IEnumerable<long>> AddBatch(List<BiasPoliticalPartiesModel> models)
+    {
+        IEnumerable<long> allInsertedIds = new List<long>();
+        _logger.LogDebug("Adding {count} political parties to database", models.Count);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        con.Open();
+        using IDbTransaction transaction = con.BeginTransaction();
+        try
+        {
+            // Divide the list of models into chunks to keep the INSERT statements from getting too large.
+            foreach (BiasPoliticalPartiesModel[] chunk in models.Chunk(_sqlHelper.InsertStatementChunkSize))
+            {
+                string parameterString = _sqlHelper.GetBatchInsertParameters(chunk, out Dictionary<string, dynamic> parameters);
+                IEnumerable<long> insertedIds = await transaction.QueryAsync<long>(
+                    $"insert into political_parties ({BiasPoliticalPartiesMap.Name}, {BiasPoliticalPartiesMap.Bias}) " +
+                    $"values {parameterString} returning {BiasPoliticalPartiesMap.Id}",
+                    parameters);
+                allInsertedIds = allInsertedIds.Concat(insertedIds);
+            }
+
+            transaction.Commit();
+        }
+        catch (Exception e)
+        {
+            transaction.Rollback();
+            _logger.LogError(e, "Failed to insert political parties");
+            throw;
+        }
+        return allInsertedIds;
+    }
+
+    public async Task<int> Add(BiasPoliticalPartiesModel entity)
+    {
+        _logger.LogDebug("Adding political party with id {partyId} to database", entity.Id);
+        _logger.LogTrace("Political Party: {party}", entity);
+        using IDbConnection con = _connectionFactory.CreateConnection();
+        return await con.ExecuteAsync(
+            $"instert into political_parties ({BiasPoliticalPartiesMap.Name}, {BiasPoliticalPartiesMap.Bias}) " +
```

Comment:
stavefejl

Reply:
_i f***ing knew it_

Reply:
Damn, Sarmi er bedre til at finde ting end ctrl-F
```suggestion
            $"insert into political_parties ({BiasPoliticalPartiesMap.Name}, {BiasPoliticalPartiesMap.Bias}) " +
```